src/lib/datatypes/include/sol-str-slice.h

```diff
@@ -123,6 +124,43 @@ sol_str_slice_to_string(const struct sol_str_slice slice)
     return strndup(slice.data, slice.len);
 }
 
+static inline struct sol_str_slice
+sol_str_slice_remove_leading_whitespace(struct sol_str_slice slice)
+{
+    struct sol_str_slice copy = slice;
+
+    while (copy.len && isspace(*copy.data)) {
+        copy.data++;
+        copy.len--;
+    }
+
+    return copy;
+}
+
+static inline struct sol_str_slice
+sol_str_slice_remove_trailing_whitespace(struct sol_str_slice slice)
+{
+    struct sol_str_slice copy = slice;
+    long idx = copy.len;
```

Comment:
then use `ssize_t` (signed) and check for overflows. I'd still follow @lpereira's advice and use `size_t`, adapting checks to not go to underflow case.